ocaml/networkd/lib/network_utils.ml

```diff
@@ -226,7 +226,9 @@ module Sysfs = struct
   let get_pci_ids name =
     let read_id_from path =
       try
-        let l = read_one_line path in
+        let l =
+          Unixext.string_of_file path |> Astring.String.take ~sat:(( <> ) '\n')
```

Comment:
This copies what `read_one_line` does, without logging on error. Another option is to add an optional argument `?(log = true)` to `read_one_line` to control whether to log or not, so we don't need to duplicate the function.

Reply:
Indeed I'm a bit puzzled about that. I understand the change proposed. On the other side this `read_one_line` function is doing something "unexpected" with `EINVAL` error. I understand it's useful for some virtual files (like reading `speed` or `carrier` which can be read only when card is up). Potentially also that `log` option (OT: maybe `ignore_errors` should be a better name?) could be reused somewhere else but it makes the `read_one_line` more complicated. In different paths in the same file something different like `Unixext.string_of_file path |> String.trim` is used.

I was also wondering if we should only ignore some type of errors, not all. For instance `EPERM` or `ENFILE` although unlikely should be reported.

One other comment on this function is that it should return either `("<VENDOR>", "<PRODUCT>")` or `("", "")`, not something like `("<VENDOR>", "")`.

My knowledge about Ocaml tends to zero.

Reply:
To ensure that last property you can move where 'try/with' is, if you move it one level up to the 'get_pci_ids' function and on error have it return `"", ""`, it should ensure the consistency property you're looking for.

Reply:
In general, I think the (existing) code is too complicated. Reading a few words from `/sys` could be simpler:

* https://github.com/xapi-project/xen-api/blob/master/ocaml/xenopsd/pvs/pvs_proxy_setup.ml#L102-L105
* https://github.com/xapi-project/xen-api/blob/master/ocaml/xenopsd/pvs/pvs_proxy_setup.ml#L259-L269
* Using scanf can do all the extra splitting and whatever you might need